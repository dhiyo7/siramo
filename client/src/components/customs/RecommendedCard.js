import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  View,
  FlatList,
  Dimensions,
  Modal,
  TouchableHighlight,
  Button,
  WebView
} from 'react-native';
import { Ionicons } from '@expo/vector-icons'

class RecommendedCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modalInfo: false,
      modalCheckmark: false,
      modalWarning: false,
      modalDanger: false
    };
  }
  render() {
    return (
      <View style={styles.mainContainer}>
          <View style={styles.iconContainer}>
            <TouchableHighlight onPress={() => this.setState({modalInfo: true})}><Ionicons name="md-information-circle" size={48} color="green"/></TouchableHighlight>
          </View>
          <View style={styles.iconContainer}>
           <Ionicons name="md-checkmark-circle" size={48} color="green" />
          </View>
          <View style={styles.iconContainer}>
           <Ionicons name="md-warning" size={48} color="yellow" />
          </View>
          <View style={styles.iconContainer}>
            <Ionicons name="md-close-circle" size={48} color="red" />
          </View>
          <Modal 
            visible={this.state.modalInfo}
            animationType={'slide'}
            transparent={true}
            onRequestClose={() => {console.log('close modal')}}
          >
            <View style={styles.modalBackGround}>
              <View style={styles.ActivityIndicatorWrapper}>
                <Text>A</Text>
                <TouchableHighlight onPress={() => this.setState({modalInfo: false})}><Text>Button</Text></TouchableHighlight>
              </View>
            </View>
          </Modal>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  mainContainer: {
    marginBottom: 4,
    marginTop: 4,
    width: 400,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    alignItems: 'center',
  },
  container:{
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    alignItems: 'center',
  },
  iconContainer:{
    justifyContent: 'center',
    alignItems: 'center',
    width: 50,
    margin: 10
  },
  descContainer: {
    width: 200,
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  warning:{
    color:'red',
    textAlign: 'center'
  },
  text:{
    textAlign: 'center'
  },
  modalBackGround: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#C8E6C9',
  },
  ActivityIndicatorWrapper: {
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    height: '100%'
  }
})

export default RecommendedCard;