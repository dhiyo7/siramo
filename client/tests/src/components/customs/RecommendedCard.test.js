import React from 'react'
import {
  View, Text, TouchableOpacity, 
  TextInput, StyleSheet, Picker
} from 'react-native'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import RecommendedCard from '../../../../src/components/customs/RecommendedCard'
import { Ionicons } from '@expo/vector-icons'

Enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<RecommendedCard />)
})

describe('<RecommendedCard /> snapshot Testing', () => {
  it('snapshot works!', () => {
    const tree = renderer.create(<RecommendedCard />)
    expect(tree).toMatchSnapshot()
  })
})

describe('<RecommendedCard />', () => {
  it('should render <RecommendedCard />', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<RecommendedCard /> state Testing', () => {
  it('should have false state value', () => {
    expect(wrapper.state('modalInfo')).toEqual(false)
    expect(wrapper.state('modalCheckmark')).toEqual(false)
    expect(wrapper.state('modalWarning')).toEqual(false)
    expect(wrapper.state('modalDanger')).toEqual(false)
  })
})

describe('<RecommendedCard /> childern Testing', () => {
  it('should render all childern', () => {
    expect(wrapper.find(<View />)).toBeDefined()
    expect(wrapper.find(<Ionicons />)).toBeDefined()
    expect(wrapper.find(<Text />)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(3)
  })
})