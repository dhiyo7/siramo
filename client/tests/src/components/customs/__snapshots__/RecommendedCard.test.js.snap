// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecommendedCard /> snapshot Testing snapshot works! 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "sans-serif-condensed",
            "textAlign": "center",
          }
        }
      >
           
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "sans-serif-condensed",
            "textAlign": "center",
          }
        }
      >
           
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "red",
            "fontFamily": "sans-serif-condensed",
            "textAlign": "center",
          }
        }
      >
           
      </Text>
    </View>
  </View>
</View>
`;
